docs/theme/mlflow/static/js/custom.js

```diff
@@ -227,3 +227,128 @@ fetch('https://pypi.org/pypi/mlflow/json')
   .catch((error) => {
     console.error('Failed to fetch package metadata from PyPI:', error);
   });
+
+// Floating side bar navigable table of contents with current document h1, h2, h3 content listings
+document.addEventListener("DOMContentLoaded", function() {
+
+    // disable short-circuit for pages where a ToC doesn't make sense
+    if (document.querySelector('.no-toc')) {
+        debugLog('ToC disabled on this page.');
+        return; 
+    }
+
+    var tocContainer = document.createElement('div');
+    tocContainer.id = 'floating-toc-container';
+    document.body.appendChild(tocContainer);
+
+    var toc = document.createElement('ul');
+    toc.id = 'floating-toc';
+    tocContainer.appendChild(toc);
+
+    var headings = document.querySelectorAll('[itemprop="articleBody"] h1, [itemprop="articleBody"] h2, [itemprop="articleBody"] h3');
```

Comment:
To support giving a table of contents in API docs, we would have to modify this query to include the itemprop entries of "python class", "python function", "python method", etc. To reuse the same CSS, we could remap these to h2 elements for the purposes of display. 
However, I didn't test this. This might be good for a followup PR later on.